### PR TITLE
[SPARK-44671][PYTHON][CONNECT] Retry ExecutePlan in case initial request didn't reach server in Python client

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

The fix for the symmetry to https://github.com/apache/spark/pull/42282.

### Why are the changes needed?

See also https://github.com/apache/spark/pull/42282

### Does this PR introduce _any_ user-facing change?

See also https://github.com/apache/spark/pull/42282

### How was this patch tested?

See also https://github.com/apache/spark/pull/42282

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @juliuszsompolski @zhengruifeng @ueshin Please take a look 🙏

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/errors/exceptions/connect.py [hunk 1]
```diff
@@ -188,3 +188,10 @@ class SparkUpgradeException(SparkConnectGrpcException, BaseSparkUpgradeException
     """
     Exception thrown because of Spark upgrade from Spark Connect.
     """
+
+
+class SparkConnectRetryException(SparkConnectException):
+    """
+    An exception that can be thrown upstream when inside retry and which will be retryable
+    regardless of policy.
+    """
```
Comment (juliuszsompolski) [inline python/pyspark/errors/exceptions/connect.py:197]: nit: if this is an internal exception that is being thrown only inside Retrying, and not an user facing exception, should this be placed somewhere next to the AttemptManager and Retrying impl?

**Review by juliuszsompolski — APPROVED** [review]
LGTM, one nit about where the internally used exception should be placed (because I don't know python style convention)

**Review by dongjoon-hyun — APPROVED** [review]

File: python/pyspark/errors/exceptions/connect.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/errors/exceptions/connect.py:197]: Actually yeah, fair enough.

**Pull request closed without merging**